sklearn/tests/test_base.py

```diff
@@ -691,3 +694,25 @@ def transform(self, X):
     # transform on feature names that are mixed also warns:
     with pytest.warns(FutureWarning, match=msg) as record:
         trans.transform(df_mixed)
+
+
+def test_base_estimator_pickle_version():
+    """Check that the original sklearn version with which a base estimator
+    has been pickled with is present"""
+    old_ver = sklearn.__version__
+    pickle_ver = old_ver + ".test1"
+    original_estimator = MyEstimator()
+
+    first_pickle_estimator = pickle.loads(pickle.dumps(original_estimator))
+    assert hasattr(first_pickle_estimator, "_sklearn_pickle_version")
+    assert first_pickle_estimator._sklearn_pickle_version == old_ver
+
+    # Change Sklearn version to simulate repickling with a newer version.
+    # Kinda hacky so if anyone knows another way to test that feel free to change this
```

Comment:
One can use pytest's `monkeypatch` fixture to override the version:

```python
def test_base_estimator_pickle_version(monkeypatch):
    """Check that the current runtime version is saved when pickling an estimator."""
    old_pickle_version = "0.21.3"
    monkeypatch.setattr(sklearn.base, "__version__", old_pickle_version)
    original_estimator = MyEstimator()

    first_pickle_estimator = pickle.loads(pickle.dumps(original_estimator))
    assert first_pickle_estimator._sklearn_pickle_version == old_pickle_version

    # Pickling the first_pickle_estimator again will update _sklearn_pickle_version to
    # the new version
    new_version = "1.1.0"
    monkeypatch.setattr(sklearn.base, "__version__", new_version)
    second_pickle_estimator = pickle.loads(pickle.dumps(first_pickle_estimator))
    assert second_pickle_estimator._sklearn_pickle_version == new_version
```

Reply:
Thanks for the tip, I modified the tests to use monkeypatch (and to take the version mismatch warning into account).